framework/web/Request.php

```diff
@@ -162,7 +162,51 @@ class Request extends \yii\base\Request
      * @see getBodyParams()
      */
     public $parsers = [];
+    /**
+     * @var array The configuration for trusted security related headers.
+     * An array key is a regular expression for matching a hostname.
+     * A value is a list of headers to trust.
+     *
+     * For example, to trust all headers from domains ending in '.trusted.com' use the following:
+     *
+     * ```php
+     * [
+     *     '/^.*\.trusted\.com$/',
+     * ```
+     *
+     * To trust just the x-forwarded-for header from domains ending in '.partial.com' use:
+     *
+     * ```
+     *     // ...
+     *     '/^.*\.partial\.com$/' => ['X-Forwarded-For']
+     * ]
+     * ```
 
+     * Default is to trusts all headers from all hosts.
+     */
+    public $trustedHostConfig = [
+        '//'
+    ];
+    /**
+     * Lists headers that are subject to the trusted host configuration.
+     * @see https://en.wikipedia.org/wiki/List_of_HTTP_header_fields
+     * @var array
+     */
+    public $secureHeaders = [
+        'X-Forwarded-For',
```

Comment:
Note that the default configuration does mean that all hosts are trusted by default. It might make sense to instead just leave the default configuration empty (thus not trusting anyone).
Since in that it could be considered a security bug that it ever used untrusted headers for security related information.

It will mean that people who trusted the `https` headers blindly and used them for redirecting will need to reconfigure.